src/main/java/com/google/devtools/build/lib/rules/proto/ProtoCompileActionBuilder.java

```diff
@@ -596,13 +605,27 @@ static void addIncludeMapArguments(
    * </code>
    */
   private static String getPathIgnoringRepository(Artifact artifact) {
+    // FIXME(jmillikin): How can I plumb `ruleContext` through the @AutoCodec closures
```

Comment:
`RuleContext` should not be put into a `TransitiveInfoProvider`. Its lifetime is limited to the analysis of that particular configured target. Why do you need to do that?